### ISSUE TITLE
No IP Shown // Can't Log In

### ISSUE THREAD

**Issue by Smoz** [xer0design/led-matrix-controller#1]
I ran the setup and after the reboot there was no IP shown so I opened a terminal, ran "hostname -I", went to the IP, saw the Apache success page only. I tried appending "/login.php" onto the IP and it showed the PHP file content (understandable :/). I then appended "/index.php" to the IP and it took me to the login page. I typed in "password" for the password but it's just going to the login.php and printing the PHP file contents as if I typed it in the first time.

Ideas?

Edit: Just tried browsing directly to "/sender.php" and it allowed me to get there. It even acted like it was going to work but no matter what I type in for the text or animations it doesn't updated the LED panels, nor does it hold the settings on the page.

Additionally, I should mention that I set this up by manually running "setup.sh" as the automated setup is throwing an error.

```
bash: line 31: syntax error near unexpected token `then'
bash: line 31: `then'
```

**Comment by xadacka** [comment]
Login.php is disabled on the default install! If you go to sender.php that's how you can control the screen

**Comment by xadacka** [comment]
Sorry, just noticed your edit. The setup script is designed to be piped into bash so just running it may have caused issues. What error was being thrown up when you piped it?

**Comment by Smoz** [comment]
I've tried a new install, removed all the files associated with this from the home/pi/ dir. From there I'm running the command you suggested from the readme:

`curl -sSL https://raw.githubusercontent.com/xer0design/led-matrix-controller/master/setup.sh | bash`

I'm getting this:

![image](https://user-images.githubusercontent.com/25465045/32509459-f9134940-c3b2-11e7-8fae-e14f455fb300.png)

After this, if I do "ls" I can see the led-matrix-controller directory, it looks like the files are in there, if I run the setup.sh like this, `sudo ./setup.sh`, then it seems to install and deploy fine. If I run the setup.sh like this, `sudo ./setup.sh | bash`, then I get this:

![image](https://user-images.githubusercontent.com/25465045/32509685-ae09a1be-c3b3-11e7-9a99-a2755e8e64da.png)

I could be doing everything wrong, this is my first project with a pi. :/

**Comment by xadacka** [comment]
I see the issue now actually, there's a syntax error in the installer, I'll fix it shortly and update here when ready to go. The installer does a lot of things and is only intended to be run once so just to be safe I'd recommend going with a fresh sd card before trying it again when fixed; code is fully tested and functional with jessie, haven't tried it on stretch yet but I'd imagine it'll work.

**Comment by xadacka** [comment]
That should be fixed and working now, the error with the webpage now loading should be fixed too! Just run the command again and leave her run. The IP showing initially I need to fix too, but that'll come soon.

curl -sSL https://raw.githubusercontent.com/xer0design/led-matrix-controller/master/setup.sh | bash

Running it as root might be a good idea, too, so type "sudo su" and return before piping.

**Comment by Smoz** [comment]
Ah ok :) I didn't see Raspbian Jessie Lite. I'll give this a try now and let you know what happens!

**Comment by Smoz** [comment]
Hmm, getting this issue now. I admittedly didn't start from scratch but I'll give that a try. It's getting to the [Y/n] choice but just aborting before I can choose anything.

![image](https://user-images.githubusercontent.com/25465045/32516807-36e6dd2e-c3ca-11e7-8583-60d0e0de7ff0.png)

**Comment by xadacka** [comment]
Okay you're still seeing a cached version of that script so, as I've removed the y/n option. I'd say give it another few minutes for the cache in your location to update. You'll know you have the correct version when it doesn't ask to update. Check the actual file in a browser and the code should jump straight from cd led-matrix-controller to #install apache2

https://raw.githubusercontent.com/xer0design/led-matrix-controller/master/setup.sh

**Comment by Smoz** [comment]
Hmm, it's strange. It's updated when I look at it in the browser but it's still doing the same thing on the pi. I'll give it a bit longer and report back.

Thanks for the quick help!

**Comment by xadacka** [comment]
I'd suggest flashing a fresh image while you wait for that

**Comment by Smoz** [comment]
Yeah I did, I'll do another ;)

**Comment by Smoz** [comment]
Any ideas about these errors? Should I just run the apt-get update --fix-missing?

![image](https://user-images.githubusercontent.com/25465045/32518200-792b3fdc-c3ce-11e7-94a0-7929a3a56fc8.png)

**Comment by Smoz** [comment]
I think that may have done it :)

**Comment by xadacka** [comment]
Yeah running that should fix it, being a brand new install it probably has no apt cache so that'd explain not being able to find anything. A simple 'apt update' should also suffice. I'll add that into the installer now too.

**Comment by Smoz** [comment]
Ugh fail :( Is this really still pulling the cached one somehow? I did get farther than I have previously! It seems like the rpi-rgb-led-matrix/examples-api-use directory isn't being created/pulled.

![image](https://user-images.githubusercontent.com/25465045/32518450-368e3994-c3cf-11e7-8b3c-c28e7266f179.png)

**Comment by xadacka** [comment]
Yeah, that question was entirely removed so it's still loading a cached copy somehow. The fact it's stating that it's making required programs without the directory there means git probably isn't installed either? I'll add that in too. 2 minutes I'll make a quick easy fix to get you by for now

**Comment by xadacka** [comment]
```
curl -sSL http://xer0.design/led/ledsetup.sh | bash
```

Try that.

**Comment by Smoz** [comment]
Oh gosh I feel stupid :/

So apparently Git wasn't installed...I just installed it and now it's able to clone the repo like it should be. I'm sorry. I'm a failure. I'll just jump off a cliff lol

**Comment by xadacka** [comment]
We all start somewhere, not to worry.

**Comment by Smoz** [comment]
I just used the latest curl you sent and it gave the same output.

![image](https://user-images.githubusercontent.com/25465045/32519056-176bebe0-c3d1-11e7-8871-647d07fdcc3b.png)

**Comment by Smoz** [comment]
Ok, so if I go into the led-matrix-controller directory and run the `./setup.sh` command then it works. It doesn't work `./setup.sh | bash` though.

It got to the [y/n] choice and waited instead of aborting immediately.

**Comment by xadacka** [comment]
Err, try one last time there, this one should do it. Use the curl from my server.

**Comment by Smoz** [comment]
Ok it went all the way through this time. Restarting now.

This is something strange that I noticed happening. I see this output stating "php5 has no installation candidate". Have you seen that? I looked it up and it says that's because "ubuntu comes with php7 installed" or something like that.

So, before I was seeing the apache webpage when I would browse to the IP, now I'm seeing that the site can't be reached because the IP refused to connect. lol

Ok just a minute I'm installing some php stuff now. 

Alright, I ran `sudo apt-get install php` and I can see the apache page once again but now instead of connection refused I get the "404 Not Found" message if I try to browse to /login.php /sender.php /index.php

**Comment by Smoz** [comment]
The web interface is now up. I rebooted it and it's up :)

Trying to do anything on the web page like update the text or anything throws an error "sorry there was an error uploading your file".

**Comment by xadacka** [comment]
Raspbian is based on Debian but it's not related to Ubuntu, it's weird to see an Ubuntu related error. If it's running anyway then no big deal.

**Comment by Smoz** [comment]
Hey! Sorry for the delay, I just got back to my pi. According to the web interface this is the current string:

![image](https://user-images.githubusercontent.com/25465045/32557151-f44adafe-c466-11e7-84b7-d36373fb9cb5.png)

But nothing is updating, just getting the error on the web form that says "sorry there was an error uploading your file". Ideas? I'm going to try a fresh install all the way through again.

**Comment by Smoz** [comment]
So just for good measure I downloaded the hzeller original to run some of the examples and make sure it wasn't my hardware. The examples run normally from the hzeller rpi-rgb-led-matrix/ directory but manually trying to run the demos in the led-matrix-controller/ rpi-rgb-led-matrix/examples-api-use/ directory don't produce any output.

So the command I'm running is:
 
`sudo ./demo --led-rows=16 --led-chain=2 --led-gpio-mapping=adafruit-hat -D 1 runtext16.ppm`

So I'm at a loss :(
